independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CreationalContextImpl.java

```diff
@@ -24,6 +24,9 @@ public class CreationalContextImpl<T> implements CreationalContext<T>, Function<
     private final CreationalContextImpl<?> parent;
     private List<InstanceHandle<?>> dependentInstances;
 
+    // may carry a current injection point or a current decorator delegate
+    private Object extra;
```

Comment:
As pointed out in the Zulip discussion, you could technically share the same `CreationalContextImpl` instance across threads which then raises a question a thread safety of the set/unset methods used here.
But take it with a pinch of salt as you'd need to go several extra miles and have a strong urge to shoot yourself in the foot to even get to that scenario :)

Reply:
Good point. I've made synchronized the accesses here.